drivers/ieee802154/ieee802154_mcxw_utils.c

```diff
@@ -0,0 +1,350 @@
+}
+
+static uint8_t CalculateAddrFieldSize(uint16_t fcf)
+{
+	if (!fcf) {
+		return 0;
+	}
+
+	uint8_t size = 2;
+
+	if (!IsSequenceNumberSuppression(fcf)) {
+		size += 1;
+	}
+
+	if (IsDstPanIdPresent(fcf)) {
+		size += 2;
+	}
+
+	// destination addressing mode
+	switch (fcf & MaskDstAddrMode) {
+	case ModeDstAddrShort:
+		size += 2;
+		break;
+	case ModeDstAddrExt:
+		size += 8;
+		break;
+	default:
+		break;
+	}
+
+	if (IsSrcPanIdPresent(fcf)) {
+		size += 2;
+	}
+
+	// source addressing mode
+	switch (fcf & MaskSrcAddrMode) {
+	case ModeSrcAddrShort:
+		size += 2;
+		break;
+	case ModeSrcAddrExt:
+		size += 8;
+		break;
+	default:
+		break;
+	}
+
+	return size;
+}
+
+static uint8_t GetKeyIdentifierMode(uint8_t *pdu, uint16_t lenght)
+{
+	uint16_t fcf = GetFrameControlField(pdu, lenght);
+	uint8_t ash_start;
+	if (IsSecurityEnabled(fcf)) {
+		ash_start = CalculateAddrFieldSize(fcf);
+		return (uint8_t)((pdu[ash_start] & MaskKeyIdMode) >> OffsetKeyIdMode);
+	}
+
+	return 0xFF;
+}
+
+bool IsKeyIdMode1(uint8_t *pdu, uint16_t lenght)
+{
+	uint8_t key_mode = GetKeyIdentifierMode(pdu, lenght);
+	if (key_mode == 0x01) {
+		return true;
+	}
+
+	return false;
+}
+
+void SetFrameCounter(uint8_t *pdu, uint16_t lenght, uint32_t fc)
+{
+	uint16_t fcf = GetFrameControlField(pdu, lenght);
+	if (IsSecurityEnabled(fcf)) {
+		uint8_t ash_start = CalculateAddrFieldSize(fcf);
+		uint8_t scf = pdu[ash_start];
+
+		// check that Frame Counter Suppression is not set
+		if (!(scf & MaskFrameCntSuppression)) {
+			sys_put_le32(fc, &pdu[ash_start + 1]);
+		}
+	}
+}
+
+static uint8_t GetASNSize(uint8_t *pdu, uint16_t lenght)
+{
+	uint16_t fcf = GetFrameControlField(pdu, lenght);
+	if (IsSecurityEnabled(fcf)) {
+		uint8_t ash_start = CalculateAddrFieldSize(fcf);
+		uint8_t scf = pdu[ash_start];
+		// scf size
+		uint8_t size = 1;
+
+		// Frame Counter Suppression is not set
+		if (!(scf & MaskFrameCntSuppression)) {
+			size += 4;
+		}
+
+		uint8_t key_mode = GetKeyIdentifierMode(pdu, lenght);
+		switch (key_mode) {
+		case 0x01:
+			size += 1;
+			break;
+		case 0x02:
+			size += 5;
+		case 0x03:
+			size += 9;
+		default:
+			break;
+		}
+
+		return size;
+	}
+	return 0;
+}
+
+static uint8_t *GetCslIeContentStart(uint8_t *pdu, uint16_t lenght)
+{
+	uint16_t fcf = GetFrameControlField(pdu, lenght);
+	if (IsIEPresent(fcf)) {
+		uint8_t ie_start_idx = CalculateAddrFieldSize(fcf) + GetASNSize(pdu, lenght);
+		uint8_t *cur_ie = &pdu[ie_start_idx];
+
+		uint8_t ie_header = (uint16_t)(cur_ie[0] | (cur_ie[1] << 8));
+		uint8_t ie_lenght = ie_header & 0x7F;
+		uint8_t ie_el_id = ie_header & 0x7F80;
+
+		while ((ie_el_id != 0x7e) && (ie_el_id != 0x7f)) {
+			if (ie_el_id == 0x1a) {
+				return (cur_ie + 2);
+			}
+			cur_ie += (2 + ie_lenght);
+			ie_header = (uint16_t)(cur_ie[0] | (cur_ie[1] << 8));
+			ie_lenght = ie_header & 0x7F;
+			ie_el_id = ie_header & 0x7F80;
+		}
+	}
+
+	return NULL;
+}
+
+void SetCslIe(uint8_t *pdu, uint16_t lenght, uint16_t period, uint16_t phase)
```

Comment:
General note - Zephyr code should be written in snake_case: https://docs.zephyrproject.org/latest/contribute/style/code.html

Reply:
Done. Thanks

Reply:
Is it? I see all the functions still use PascalCase.

Reply:
Yes, sorry. Now it's done